scripts/backup-restore-test.sh

```
@@ -291,97 +371,32 @@ verify_fk_integrity() {
-
-    if [ -n "$orphans_detected" ] && [ "$orphans_detected" -gt 0 ]; then
-        log_warn "Found $orphans_detected orphaned row(s) across FK constraints."
-        report_metric "fk_orphans" "$orphans_detected"
-    else
-        log_ok "No orphaned rows detected — FK referential integrity intact."
-        report_metric "fk_orphans" "0"
-    fi
+    # Check for actual orphaned rows using a safe heuristic:
+    # sample a subset of referenced values in the largest FK
+    local orphan_found
+    orphan_found=$(psql "$target_url" -X -t -A -c "
```

Shellcheck (0.11.0)

[warning] orphan_found appears unused. Verify use (or export if used externally).

(SC2034)